src/themes/custom_theme/custom_theme.js

```diff
@@ -0,0 +1,246 @@
+
+		var rgba = [];
+		rgba[0] = parseInt(value.substring(0,2),16),
+		rgba[1] = parseInt(value.substring(2,4),16),
+		rgba[2] = parseInt(value.substring(4,6),16),
+		rgba[3] = alpha? parseFloat(alpha) : 1;
+		var temp = "",
+				end = "";
+		var propertyNames = id.split("_");
+		propertyNames.forEach(function(property, index, arr){
+			// For creating array.
+			if(property === "arr"){
+				temp = temp + "[" + "\"rgba(" + rgba.join(", ") + ")\"" + "]";
+				return;
+			}
+			end = end + "}";
+			if(index === arr.length - 1){
+				temp = temp + "{" + "\"" + property + "\"" + ":" + "\"rgba(" + rgba.join(", ") + ")\"";
+				return;
+			}
+			temp = temp + "{" + "\"" + property + "\"" + ":";
+		});
+		temp = temp + end;
+		$.extend(themeConf, JSON.parse(temp));
+	}
+
+	$('a.theme_custom')
+    .off('click')
+    .on('click', function(){
+
+    	var ele = $(this);
+
+    	require(['css!themes/custom_theme/custom_theme.css']);
+
+      require(['text!themes/custom_theme/custom_theme.html'], function($html){
+      	$html = $($html);
+
+      	$html.find(".color_input_width").each(function(index, value){
+      		var id = $(value).attr("id").replace("theme_",""),
+      				alpha = $(value).attr("alpha");
+      		$(value).colorpicker({
+      			part:	{
+                    map:		{ size: 128 },
+                    bar:		{ size: 128 }
+                },
+                select:	function(event, color) {
+                    $(value).css({
+                        background: '#' + color.formatted
+                    }).val('');
+                    createThemeObject(id, color.formatted, alpha);
+										updateChart();
+                },
+                ok: function(event, color) {
+                    $(value).css({
+                        background: '#' + color.formatted
+                    }).val('');
+                    createThemeObject(id, color.formatted, alpha);
+                    updateChart();
+                }
+      		});
+      	});
+
+      	win = windows.createBlankWindow($html,
+      		{
+      			autoOpen: false,
+            resizable: false,
+            collapsable: false,
+            minimizable: false,
+            maximizable: false,
+            closable: false,
+            closeOnEscape: false,
+            width: 650,
+            height: 515,
+            title: "Customize chart appearance",
+            modal: true,
+            my: 'center',
+            at: 'center',
+            of: window,
+            destroy: function() {
+                win = null;
+            },
+            buttons: {
+              Apply: function() {
+              		openConfirmDialog();
+              },
+              Cancel: function() {
+                  $( this ).dialog( 'close' );
+                  $( this ).dialog( "destroy" );
+                  var themeName = local_storage.get("theme");
+                  themeName = themeName && themeName.name;
+                  //Reseting charts to previous theme or default.
```

Comment:
Yes @arnabk  this is necessary because once the highcharts options are set for one chart it automatically applies to all new chart which will be created after execution of the Highcharts.setOptions() statement.